### DISCUSSION TITLE
Should checkstyle be executed in the fastest build profile?

### DISCUSSION THREAD

**Discussion by escay** [eclipse-ee4j/glassfish#25130] [Q&A] [answered]
Hi,

when I build with
`mvn clean install -Pfastest -T4C` or 
`mvn clean install -Pfast`
it seems like the checkstyle plugin is not executed. 

Is this on purpose to make the fastest build as fast as possible?
I thought let's build using fastest to see if my merge was correct.. 

Workaround: build for example using:
`mvn clean install -Pqa`

**Comment by pzygielo** [comment, marked as answer]
> when I build with `mvn clean install -Pfastest -T4C` or `mvn clean install -Pfast` it seems like the checkstyle plugin is not executed.
> 
> Is this on purpose to make the fastest build as fast as possible?

Seems to be on purpose, given `checkstyle.skip=true` is set in

https://github.com/eclipse-ee4j/glassfish/blob/832cf832d2caaba69941453c8a76940c96fc3609/nucleus/parent/pom.xml#L1522-L1527

https://github.com/eclipse-ee4j/glassfish/blob/832cf832d2caaba69941453c8a76940c96fc3609/nucleus/parent/pom.xml#L1530-L1535

**Reply by escay** [reply]
Thank you for the pointers to the code. I was wondering from a development perspective what would be expected from the profiles:
In my case I am building most times because I altered some code, and therefor I assumed checkstyle is always enabled in any profile.

**Reply by OndroMih** [reply]
You can make it enabled all the time if you set the checkstyle.skip property in settings.xml to true. Or you can add .mvn/maven.config into your local GF repo clone and specify the property there, if tou want to set it only for GF builds

**Reply by OndroMih** [reply]
Checkstyle makes the build slower. It doesn’t make sense to run it for the code that didn’t change. I usually build changed modules from IDE (Netbeans), without the fastest profile, so the IDE reports checkstyle errors. If a module is built OK, I run full build with fastest profile

**Reply by dmatej** [reply]
fastest profile's target is to get distribution artifacts asap.
fast should run also unit tests which are still quite fast.
On my computer it is like 90 seconds vs 5 minutes vs 25 minutes (clean verify without explicit profiles).

**Reply by dmatej** [reply]
Yeah, I have these profiles in my local settings.xml for years too, they are useful also for other projects.
